### DISCUSSION TITLE
Data Restore is not working using shell

### DISCUSSION THREAD

**Discussion by ankitsoin** [finos/TimeBase-CE#270] [General]
I did exactly same as mentioned you in the link : https://github.com/finos/TimeBase-CE/discussions/226 
I can see zip created on defined path but while importing data back to the stream it shows exception as below.

set raw on
set stream test_stream
export-with-spaces dirA/subdir/data.zip

open rw
set src dirA/subdir/data.zip
set stream test_stream
import

==> import
Importing; hit <Enter> to abort ...
>>> Error: java.lang.UnsupportedOperationException: Old file format = 0 is not supported.
        at com.epam.deltix.qsrv.hf.stream.Protocol.readHeader(Protocol.java:179)
        at com.epam.deltix.qsrv.hf.tickdb.ui.tbshell.ImportExportHelper.filterMessageFile(ImportExportHelper.java:112)
        at com.epam.deltix.qsrv.hf.tickdb.ui.tbshell.TickDBShell.doCommand(TickDBShell.java:414)
        at com.epam.deltix.util.cmdline.AbstractShell.runCommand(AbstractShell.java:214)
        at com.epam.deltix.util.cmdline.AbstractShell.runScript(AbstractShell.java:374)
        at com.epam.deltix.util.cmdline.AbstractShell.run(AbstractShell.java:302)
        at com.epam.deltix.util.cmdline.DefaultApplication.start(DefaultApplication.java:445)
        at com.epam.deltix.qsrv.hf.tickdb.ui.tbshell.TickDBShell.main(TickDBShell.java:94)

**Comment by alex-karpovich** [comment]
please use command:

```
open rw
set src dirA/subdir/data.zip
import test_stream
```

**Reply by ankitsoin** [reply]
Thanks @alex-karpovich for quick help.

Couple of query :
Can we automate this process using script as not sure if we can go inside the shell and execute the commands using .sh script?
Do we have functionality where we can export only specific space data for the backup purpose(I can see we can do using symbol or timestamp in documentation) ?

Please feel free to add more on recommended appraoch for backup purpose? (I have loaded data in  the spaces of the stream)

**Reply by alex-karpovich** [reply]
Please take look at the https://kb.timebase.info/docs/development/tools/shell#command-line-options

**Reply by alex-karpovich** [reply]
you can create own script file with list of TimeBase shell commands and give it as argument to TimeBase Shell CLI

**Reply by ankitsoin** [reply]
Thanks, @alex-karpovich  it worked.
But what I can do to take backup form specific spaces for incremental backup?
If this feature is not there in community version is it there in Enterprise Edition?

**Reply by alex-karpovich** [reply]
I can add such option to CE edition on next week

**Reply by ankitsoin** [reply]
Yes please, thanks a lot.
Will be waiting for same.

**Reply by ankitsoin** [reply]
Hi @alex-karpovich, is the feature of taking backup/incremental backup for spaces is added in CE?

**Reply by ankitsoin** [reply]
HI @alex-karpovich, may I know is the feature added?
Also, is this feature is there in enterprise version?

**Reply by alex-karpovich** [reply]
I will publish tomorrow, sorry for delay

**Reply by ankitsoin** [reply]
Hi @alex-karpovich, any update please?
